### ISSUE TITLE
Might have some problems with some Web Apps?

### ISSUE THREAD

**Issue by mrdokenny** [tumpio/requestcontrol#18]
So I'm trying to use a filter rule for Facebook with these settings

host: *.facebook.com
path: *
action: filter
filterURL: On
Trim URL Parameters: Trim all

The problem is that when I navigate on to a page with a ref on facebook it seems to not be filtered (doesn't seem to do an page update event for the extension).

Ex. Entering someones profile via hovering over the their chat gives this link:
https://www.facebook.com/{UserProfile}?fref=hovercard&hc_location=chat

Target:
https://www.facebook.com/{UserProfile}

However, refreshing the page does do the filter so I know the rule is working.

Browser: FF 55
RC Version: 1.6.1

Thanks for this cool extension 😄

**Comment by tumpio** [comment]
I'll investigate it. Did you have which types checked for the rule?

**Comment by mrdokenny** [comment]
Only document.

**Comment by tumpio** [comment]
I haven't had time to check this. Can you try adding "Embedded document" and "Any type" for the rule? Let me know if it catches the request.

**Comment by mrdokenny** [comment]
I did a little bit more testing and adding Embedded document did _work_ with Trim All. Now the next problem is Trim All seems to be doing to much since FB has some legitimate uses for it. For example, users who haven't set a custom username have the following url

https://www.facebook.com/profile.php?id=123456789

So trim all would break that, but when I put fref and hc_location in the parameters, it stopped working. 

Also another thing, there may be 1 or more request for a given page on FB (embedded pages?) so it might be helpful to log all of those to see requested and target urls.

**TL:DR**
host: *.facebook.com
path: *
types: document, embedded document
action: filter
filterURL: On
Trim URL Parameters: Trim all

Works but it breaks some stuff

host: *.facebook.com
path: *
types: document, embedded document
action: filter
filterURL: On
Trim URL Parameters: fref, hc_location

Doesn't work 

When I get more time, I'll do some more testing.

**Comment by tumpio** [comment]
Facebook and other single page web applications can retrieve new content in background. Requests for documents retrieved in background are caught using the "Embedded document" type. (I think I'll rename it to sub document or sub frame). You can use Firefox Network Console to inspect what requests Facebook does in background.

Single page web apps can also update browser's url path in background and change url query string without browser actually sending any requests or parameters.

**Comment by mrdokenny** [comment]
Ok I did some more testing and per your advice I used the Network console to figure out which parameters to trim without breaking everything.

If anyone's interested here is my final FB settings:
scheme: http/https
host: *.facebook.com
path: *
types: document, sub document
action: filter
filter URL: Off
trim URL parameters: *ref, hc_location, ajaxpipe_token, dpr

I think this can be safely closed but I would recommend a tutorial for other users on how to use the Network Console (or something in the help docs that points them to the Network Console) to create rules.

**Comment by tumpio** [comment]
> I think this can be safely closed but I would recommend a tutorial for other users on how to use the Network Console (or something in the help docs that points them to the Network Console) to create rules.

I'll make a note about it in the manual. You can check tutorials for using Firefox dev tools from Mozilla docs:
https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor

**Comment by tumpio** [comment]
The manual is a markdown document here:
https://github.com/tumpio/requestcontrol/blob/master/_locales/en/manual.md

You are free to contribute changes. The wiki is also public if you want to contribute content there.
